mlflow/store/artifact/databricks_sdk_models_artifact_repo.py

```diff
@@ -0,0 +1,93 @@
+import io
+import os
+import posixpath
+
+from mlflow.entities import FileInfo
+from mlflow.store.artifact.cloud_artifact_repo import CloudArtifactRepository
+
+def _get_databricks_workspace_client():
+    from databricks.sdk import WorkspaceClient
+
+    return WorkspaceClient()
+
+
+class DatabricksSDKModelsArtifactRepository(CloudArtifactRepository):
+    """
+    Stores and retrieves model artifacts via Databricks SDK, agnostic to the underlying cloud
+    that stores the model artifacts.
+    """
+
+    def __init__(self, model_name, model_version):
+        self.model_name = model_name
+        self.model_version = model_version
+        from mlflow.utils.logging_utils import eprint
+        eprint("Using DatabricksSDKModelsArtifactRepository")
+        model_name_parts = model_name.split(".")
+        catalog = model_name_parts[0]
+        schema = model_name_parts[1]
+        model = model_name_parts[2]
+        self.model_base_path = f"/Models/{catalog}/{schema}/{model}/{model_version}"
+        super().__init__(self.model_base_path)
+
+    def list_artifacts(self, path=None):
+        dest_path = self.model_base_path
+        if path:
+            dest_path = posixpath.join(dest_path, path)
+
+        workspace_client = _get_databricks_workspace_client()
+
+        file_infos = []
+        resp = workspace_client.files.list_directory_contents(dest_path)
+        for directory_entry in resp:
+            relative_path = os.path.relpath(directory_entry.path, self.model_base_path)
+            file_infos.append(
+                FileInfo(
+                    path=relative_path,
+                    is_dir=directory_entry.is_directory,
+                    file_size=directory_entry.file_size,
+                )
+            )
+
+        return sorted(file_infos, key=lambda f: f.path)
+
+    def _upload_to_cloud(self, cloud_credential_info, src_file_path, artifact_file_path=None):
+        dest_path = self.model_base_path
+        if artifact_file_path:
+            dest_path = posixpath.join(dest_path, artifact_file_path)
+
+        with open(src_file_path, "rb") as f:
+            contents = io.BytesIO(f.read())
+
+        workspace_client = _get_databricks_workspace_client()
+        workspace_client.files.upload(dest_path, contents)
```

Comment:
Does `workspace_client.files.upload` support streaming upload?

Reply:
It does not. But there's plan for this near end of Q4
We're implementing this right now to unblock the small subset of SEG customers (better than having no mlflow registry at all)

Reply:
I think I misunderstood your question here. Yes it should support streaming upload (given the 5GB limit). Will fix the code here to take advantage of streaming.